File: src/main/java/com/bajdcc/LALR1/semantic/tracker/InstructionRecord.java
package com.bajdcc.LALR1.semantic.tracker;

import java.util.ArrayList;
import java.util.List;

/**
 * 下推自动机指令记录器链表
 *
 * @author bajdcc
 */
public class InstructionRecord {

	/**
	 * 指令集
	 */
	public List<Instruction> arrInsts = new ArrayList<>();

	/**
	 * 前向指针
	 */
	public InstructionRecord prev;

	public InstructionRecord(InstructionRecord prev) {
		this.prev = prev;
	}
}

File: src/main/java/com/bajdcc/LALR1/semantic/tracker/InstructionRecord.kt
package com.bajdcc.LALR1.semantic.tracker

/**
 * 下推自动机指令记录器链表
 * [prev] 前向指针
 * [insts] 指令集
 * @author bajdcc
 */
data class InstructionRecord(var prev: InstructionRecord?,
                             var insts: MutableList<Instruction> = mutableListOf())